* this is an incomplete patch, adding CFLAGS to some targets that we use
  without this patch, one can't configure john via defines

--- src/Makefile
+++ src/Makefile
@@ -597,7 +597,7 @@
 	$(LN) ppc32.h arch.h
 	$(MAKE) $(PROJ) \
 		CC=cc \
-		CFLAGS="-c -qunroll=2 -qarch=ppc -qchars=signed" \
+		CFLAGS="$(CFLAGS) -c -qunroll=2 -qarch=ppc -qchars=signed" \
 		LDFLAGS="-s -lbsd" \
 		OPT_NORMAL="-O2" \
 		OPT_INLINE="-O3 -Q=99 -w"
@@ -626,7 +626,7 @@
 	$(LN) ppc32alt.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) ppc-alti-macosx.o" \
-		CFLAGS="-c -Wall -fomit-frame-pointer" \
+		CFLAGS="$(CFLAGS) -c -Wall -fomit-frame-pointer" \
 		OPT_NORMAL="-fast -mcpu=7450" \
 		OPT_INLINE="-fast -mcpu=7450 -finline-limit=4000"
 
@@ -637,7 +637,7 @@
 	$(LN) ppc32.h arch.h
 	$(MAKE) $(PROJ) \
 		CC=cc \
-		CFLAGS="-c -traditional-cpp" \
+		CFLAGS="$(CFLAGS) -c -traditional-cpp" \
 		OPT_NORMAL="-O2" \
 		OPT_INLINE="-O3"
 
@@ -646,7 +646,7 @@
 	$(LN) ppc32.h arch.h
 	$(MAKE) $(PROJ) \
 		ASFLAGS="$(ASFLAGS) -arch ppc" \
-		CFLAGS="-c -Wall -arch ppc -fomit-frame-pointer" \
+		CFLAGS="$(CFLAGS) -c -Wall -arch ppc -fomit-frame-pointer" \
 		LDFLAGS="$(LDFLAGS) -arch ppc" \
 		OPT_NORMAL="-O2" \
 		OPT_INLINE="-O3 -finline-limit=4000"
@@ -656,7 +656,7 @@
 	$(LN) ppc64alt.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) ppc-alti-macosx.o" \
-		CFLAGS="-c -m64 -Wall -fomit-frame-pointer" \
+		CFLAGS="$(CFLAGS) -c -m64 -Wall -fomit-frame-pointer" \
 		LDFLAGS="$(LDFLAGS) -m64" \
 		OPT_NORMAL="-fast" \
 		OPT_INLINE="-fast -finline-limit=4000"
@@ -664,7 +664,7 @@
 macosx-ppc64:
 	$(LN) ppc64.h arch.h
 	$(MAKE) $(PROJ) \
-		CFLAGS="-c -Wall -fomit-frame-pointer -m64" \
+		CFLAGS="$(CFLAGS) -c -Wall -fomit-frame-pointer -m64" \
 		LDFLAGS="$(LDFLAGS) -m64" \
 		OPT_NORMAL="-fast" \
 		OPT_INLINE="-fast -finline-limit=4000"
@@ -698,7 +698,7 @@
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) ppc-alti-macosx.o" \
 		ASFLAGS="$(ASFLAGS) -arch ppc" \
-		CFLAGS="-c -Wall -arch ppc -fomit-frame-pointer" \
+		CFLAGS="$(CFLAGS) -c -Wall -arch ppc -fomit-frame-pointer" \
 		LDFLAGS="$(LDFLAGS) -arch ppc" \
 		OPT_NORMAL="-fast -mcpu=7450" \
 		OPT_INLINE="-fast -mcpu=7450 -finline-limit=4000"
